test/new-e2e/go.mod

```
@@ -1,8 +1,6 @@
 module github.com/DataDog/datadog-agent/test/new-e2e
 
-go 1.22.5
-
-toolchain go1.22.8
+go 1.23.0
 
```

```suggestion
go 1.23.3

toolchain go1.23.3
```
Something like that ?

---

The version in the `go.mod` is the minimal version require so `go 1.23.0` is correct.
If you want why go `1.23.0` can be better than `1.23.3` you can read [this comment](
https://github.com/golang/go/issues/62278#issuecomment-1698829945).